c2corg_ui/format/video.py

```diff
@@ -0,0 +1,70 @@
+'''
+c2corg video Extension for Python-Markdown
+==============================================
+
+Converts video tags to advanced HTML video tags.
+'''
+
+from markdown.extensions import Extension
+from markdown.inlinepatterns import Pattern
+from markdown.util import etree
+import re
+
+VIDEO_RE = r'\[video\](.*?)\[/video\]'
+
+
+class C2CVideoExtension(Extension):
+
+    def extendMarkdown(self, md, md_globals):  # noqa
+        self.md = md
+
+        pattern = C2CVideo(VIDEO_RE)
+        pattern.md = md
+        # append to end of inline patterns
+        md.inlinePatterns.add('c2cvideo', pattern, "<not_strong")
+
+
+class C2CVideo(Pattern):
+
+
+    def handleMatch(self, m):  # noqa
+        link = m.group(2).strip()
+
+        # youtube http://www.youtube.com/watch?v=3xMk3RNSbcc(&something)
+        match = re.search(r"https?:\/\/(?:www\.)?youtube\.com/watch\?(?:[=&\w]+&)?v=([-\w]+)(?:&.+)?(?:\#.*)?", link)
```

Comment:
The linter expects lines to be smaller than 80 characters. In cases where it's not possible to do so, you can tell the linter to ignore the line by adding ``# noqa`` at the end of the line. See the function name for instance (in that case the problem was the camelCase used instead of function_name but here we override a third-party lib class function).